airflow/contrib/hooks/gcp_dlp_hook.py

```diff
@@ -0,0 +1,1362 @@
+    def redact_image(
+        self,
+        parent,
+        inspect_config=None,
+        image_redaction_configs=None,
+        include_findings=None,
+        byte_item=None,
+        retry=None,
+        timeout=None,
+        metadata=None,
+    ):
+        """
+        Redacts potentially sensitive info from an image. This method has limits on
+        input size, processing time, and output size.
+
+        :param parent: The parent resource name.
+        :type parent: str
+        :param inspect_config: (Optional) Configuration for the inspector. Items specified
+            here will override the template referenced by the inspect_template_name argument.
+        :type inspect_config: dict or google.cloud.dlp_v2.types.InspectConfig
+        :param image_redaction_configs: (Optional) The configuration for specifying what
+            content to redact from images.
+        :type image_redaction_configs: list[dict] or list[google.cloud.dlp_v2.types.ImageRedactionConfig]
+        :param include_findings: (Optional) Whether the response should include findings
+            along with the redacted image.
+        :type include_findings: bool
+        :param byte_item: (Optional) The content must be PNG, JPEG, SVG or BMP.
+        :type byte_item: dict or google.cloud.dlp_v2.types.ByteContentItem
+        :param retry: (Optional) A retry object used to retry requests.
+            If None is specified, requests will not be retried.
+        :type retry: google.api_core.retry.Retry
+        :param timeout: (Optional) The amount of time, in seconds, to wait for the request
+            to complete. Note that if retry is specified, the timeout applies to each
+            individual attempt.
+        :type timeout: float
+        :param metadata: (Optional) Additional metadata that is provided to the method.
+        :type metadata: sequence[tuple[str, str]]]
+        :rtype: google.cloud.dlp_v2.types.RedactImageResponse
+        """
+
+        client = self.get_conn()
+
+        if not parent:
+            raise AirflowException("Please provide the parent resource name.")
+
+        return client.redact_image(
+            parent=parent,
+            inspect_config=inspect_config,
+            image_redaction_configs=image_redaction_configs,
+            include_findings=include_findings,
+            byte_item=byte_item,
+            retry=retry,
+            timeout=timeout,
+            metadata=metadata,
+        )
+
+    @GoogleCloudBaseHook.catch_http_exception
+    def reidentify_content(
+        self,
+        parent,
+        reidentify_config=None,
+        inspect_config=None,
+        item=None,
+        inspect_template_name=None,
+        reidentify_template_name=None,
+        retry=None,
+        timeout=None,
+        metadata=None,
+    ):
+        """
+        Re-identifies content that has been de-identified.
+
+        :param parent: The parent resource name.
+        :type parent: str
```

Comment:
It is worth adding the possibility of defining the project ID using the connection configuration.